tests/unit/TechRecordsService.unitTest.ts

```diff
@@ -657,6 +655,66 @@ describe("updateTechRecord", () => {
         }
       });
     });
+
+    context("and the user wants to update an archived record", () => {
+      it("should return error 400 Cannot update an archived record", async () => {
+        // @ts-ignore
+        const techRecord: ITechRecordWrapper = cloneDeep(records[31]);
+        const MockDAO = jest.fn().mockImplementation();
+        const mockDAO = new MockDAO();
+        const techRecordsService = new TechRecordsService(mockDAO);
+        const recordToUpdate: any = {
+          techRecord: techRecord.techRecord
+        };
+        try {
+          expect(await techRecordsService.updateTechRecord(recordToUpdate, msUserDetails)).toThrowError();
+        } catch (errorResponse) {
+          expect(errorResponse).toBeInstanceOf(HTTPError);
+          expect(errorResponse.statusCode).toEqual(400);
+          expect(errorResponse.body).toEqual(ERRORS.CannotUpdateArchivedRecord);
+        }
+      });
+    });
+
+    context("and the user wants to update a record which doesn't have the specified statusCode", () => {
+      it("should return error 500 Vehicle has no tech-records with status {statusCode}", async () => {
+        // @ts-ignore
+        const techRecord: ITechRecordWrapper = cloneDeep(records[43]);
+        const MockDAO = jest.fn().mockImplementation();
+        const mockDAO = new MockDAO();
```

Comment:
given that you don't used these named consts again, I'd probably do them inline:
`new TechRecordsService(new MockDAO());`
or even 
`new TechRecordsService(new jest.fn());` 
to skip MockDAO enirely

Reply:
I changed it to new TechRecordsService(new MockDAO()); For some reason if I wanted to use new jest.fn() typescript would complain with "'new' expression, whose target lacks a construct signature, implicitly has an 'any' type" and I couldn't find a fix for it.

Reply:
oh, brackets fix that - I initially  included them in my suggestion and then took them out
`new (jest.fn())`